### ISSUE TITLE
Body gets focus after user changes month with mouse click

### ISSUE THREAD

**Issue by frodehansen2** [airbnb/react-dates#1065]
https://github.com/airbnb/react-dates/blob/1365e9b895b424e0142cee1e9870d0e30d96aecd/src/components/DayPicker.jsx#L593

Why is it better to remove focus from an element and deviate from default behaviour, than maintain consistency? These lines which blurs the active element, prevents the user from using the arrow-keys after changing month with mouse-clicking and it removes the visual indicator of where the focus is (since body gets the focus). Is it not better to be consistent and select the first selectable day, or at least maintain focus on the button which is clicked?

We want to have good marking on the active element in the page, but with the logic as it is now, the user cannot see where he/she is (because focus is on body).

**Comment by majapw** [comment]
Thanks for opening this issue @frodehansen2!

I'm tagging @erin-doyle @backwardok for their a11y expertise.

Visible keyboard focus on the day elements has often been in conflict with the aesthetics of our designs. I'm not sure if I would necessarily say that the choice that was made there is inherently better than moving focus back to the focused day, but I *think* that is the tradeoff I was making at the time. The thought was that mouse users would likely consistently use their mouse through their interaction with the datepicker, while keyboard users would consistently be using their keyboard across the experience. Therefore, we would build for that assumption rather than trying to make the ideal experience for someone who was switching back and forth. 

We are open to suggestions! And maybe one of our resident a11y experts can advise as well. :)

**Comment by backwardok** [comment]
I found an [article](http://www.davidmacd.com/blog/sighted-keyboard-only-user.html 
) yesterday that highlights the "keyboard only user", and the biggest thing it highlights is that even for users where they predominantly navigate by keyboard, they may still use a mouse if it's easier than using a keyboard for a particular action. Not sure of what solution would be visually best for the focus handling, but I can see where having it not blur and reset to the top would be ideal

**Comment by frodehansen2** [comment]
As always there are pros and cons with all solution, but would it not be better to handle the visual with css rather than actually moving focus out of context to get an visual effect. But the the easiest fix, which still is functionally ok, is probably to keep the button focused instead of moving it to the body.
Looking forward for the a11y expert insights.

**Comment by backwardok** [comment]
Since we're doing custom focus states anyway, maybe it makes sense to just visually unfocus but leave it technically focused

**Comment by DBjelovuk** [comment]
In IE, this behavior seems to trigger focus change and closes the calendar any time the user tries to navigate the months with their mouse.

**Comment by erin-doyle** [comment]
This sounds like the same issue as: https://github.com/airbnb/react-dates/issues/1420 and hopefully could be fixed if we can finish fixing: https://github.com/airbnb/react-dates/issues/1484.  Unless I'm misunderstanding anything?

**Comment by erin-doyle** [comment]
No, I misread what this was about initially and understand it now.  I ran across this code while working on a fix for Issue #1484 and was wondering why it was there.  I think it would be reasonable to remove this block of code altogether and let the focus remain the same whether you're using the keyboard or mouse click.
